src/Handler/CurlFactory.php

```diff
@@ -375,6 +375,14 @@ private function applyHandlerOptions(EasyHandle $easy, array &$conf)
             $conf[CURLOPT_TIMEOUT_MS] = $options['timeout'] * 1000;
         }
 
+        if (isset($options['force_ipv4'])) {
+            if (defined('CURLOPT_IPRESOLVE') && defined('CURL_IPRESOLVE_V4')) {
+                $conf[CURLOPT_IPRESOLVE] = CURL_IPRESOLVE_V4;
+            } else {
+                throw new \RuntimeException('Can not force client to use IPV4 protocol on php version ' . phpversion());
```

Comment:
Can we do the check in a reverse order? If the feature is not available throw an exception else just continue running. (You don't even need the else branch)